craft_store/publishergateway/_publishergw.py

```diff
@@ -0,0 +1,107 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Client for the publisher gateway."""
+
+from typing import cast
+
+import httpx
+
+from craft_store import auth, candidauth, errors
+
+from . import _request, _response
+
+
+class PublisherGateway:
```

Comment:
Right now the import path seems pretty weird:
`craft_store.publishergateway.PublisherGateway`, should we at least rename it to `craft_store.publisher_gateway.PublisherGateway` or maybe even `craft_store.clients.PublisherGateway` ?

Reply:
Definitely not `craft_store.clients.PublisherGateway`, the namespace for the publisher gateway and the snap store dashboard are likely to have a lot of overlap.

I've renamed it to `publisher` and made it so you can either call `craft_store.PublisherGateway` (if you just want the gateway) or `craft_store.publisher.PublisherGateway` (if you want the full namespace with the TypedDicts).